### PR TITLE
Add estimation methods to RobotController

### PR DESCRIPTION
**Pull request by henneboy**
Added:

- EstimateTimeToTarget
- EstimateDistanceToTarget

### CODE REVIEW

**Review by Puvikaran2001 — APPROVED** [review]

File: Assets/Scripts/Robot/Robot2DController.cs [hunk 1]
```diff
@@ -468,6 +468,75 @@ public void PathAndMoveTo(Vector2Int tile)
...
+            }
+            else
+            {
+                dist -= distForMaxSpeed;
+                return (int)Math.Ceiling(CorrectForRelativeMoveSpeed(dist, Constraints.RelativeMoveSpeed)) + startTime;
+            }
+
+            static float CorrectForRelativeMoveSpeed(float distance, float relativeMoveSpeed)
+            {
+                return distance * 3.2f / (0.21f + (relativeMoveSpeed / 3.0f));
```
Comment (S3JER) [inline Assets/Scripts/Robot/Robot2DController.cs:502]: Add comments of what the numbers represent

File: Assets/Scripts/Robot/Robot2DController.cs [hunk 2]
```diff
@@ -468,6 +468,75 @@ public void PathAndMoveTo(Vector2Int tile)
...
+            // Is Constraints.RelativeMoveSpeed equal to the speed or is this multiplied by the movement force?
+            var distForMaxSpeed = 2.5f;
+            var distance = EstimateDistanceToTarget(target);
+            if (distance == null)
+            {
+                return null;
+            }
+            var dist = distance.Value;
+            var startDist = Math.Min(dist, 2.5f);
+            var startTime = (int)Math.Floor(Math.Pow(CorrectForRelativeMoveSpeed(startDist, Constraints.RelativeMoveSpeed), 0.85));
```
Comment (S3JER) [inline Assets/Scripts/Robot/Robot2DController.cs:489]: Maybe use constants here?

**Review by S3JER — CHANGES_REQUESTED** [review]
Check my comments

File: Assets/Scripts/Robot/Robot2DController.cs (hunk 1 above)
Comment (henneboy) [reply Assets/Scripts/Robot/Robot2DController.cs:505]: Done

File: Assets/Scripts/Robot/Robot2DController.cs (hunk 2 above)
Comment (henneboy) [reply Assets/Scripts/Robot/Robot2DController.cs:490]: Done

**Review by S3JER — APPROVED** [review]

**Pull request merged**